### PR TITLE
Riot: keep inserting data when the QED leader changes

### PR DESCRIPTION
**Pull request by jllucas**
In short:
- return error if primary is dead when asking for primary.
- layout: refactor server start/stop/clean scripts and move to "deploy/local".
- layout: move agents start/stop scripts to "deploy/local".

### CODE REVIEW

File: deploy/local/agents_start [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
```
Comment (iknite) [inline deploy/local/agents_start]: #!/user/bin env ?

File: deploy/local/agents_start [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#!/user/bin/env bash
```
Comment (gdiazlo) [inline deploy/local/agents_start]: I guess this should be /usr/bin/env

File: deploy/local/servers_start [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/user/bin/env bash
```
Comment (gdiazlo) [inline deploy/local/servers_start]: This should be /usr/bin/env

File: deploy/local/servers_stop [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env sh
```
Comment (gdiazlo) [inline deploy/local/servers_stop]: I think this should be bash, the array notation is modern bash and not sh afaik

File: deploy/local/start_all_xterm [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/user/bin/env bash
```
Comment (gdiazlo) [inline deploy/local/start_all_xterm]: This should be /usr/bin/env

File: deploy/local/agents_stop [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env sh
```
Comment (gdiazlo) [inline deploy/local/agents_stop]: This should be bash instead of sh.

File: deploy/local/agents_clean [hunk 7]
```diff
@@ -13,13 +13,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+set -x
```
Comment (suizman) [inline deploy/local/agents_clean]: Not necessary by default IMHO

File: deploy/local/servers_clean [hunk 8]
```diff
@@ -13,14 +13,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+set -x
```
Comment (suizman) [inline deploy/local/servers_clean]: Same here

**Pull request merged**